File: src/main/java/ukalus/graph/Edge.java
package ukalus.graph;

import java.util.List;

import static java.util.Arrays.asList;

public class Edge {

  private Node from;
  private Node to;

  public Edge(Node from, Node to) {
    this.from = from;
    this.to = to;
  }

  public List<Node> getNodes() {
    return asList(from, to);
  }

  public Node getNode(Node node) {
    if (node.equals(from)) {
      return to;
    } else if (node.equals(to)) {
      return from;
    }
    return null;
  }

}

File: src/main/kotlin/ukalus/graph/Edge.kt
package ukalus.graph

open class Edge(private val from: Node, private val to: Node) {

    val nodes: List<Node>
        get() = listOf(from, to)

    fun getNode(node: Node) = when (node) {
        from -> to
        to -> from
        else -> null
    }

}